src/doghouse/core/ports/github_port.py

```
@@ -0,0 +1,21 @@
+from abc import ABC, abstractmethod
+from typing import Dict, Any, List, Optional
+from ..domain.blocker import Blocker
+
+class GitHubPort(ABC):
+    """Port for interacting with GitHub to fetch PR state."""
+    
+    @abstractmethod
+    def get_head_sha(self, pr_id: Optional[int] = None) -> str:
+        """Get the current head SHA of the PR."""
+        pass
```

Ruff (0.15.7)

[warning] Unnecessary `pass` statement

Remove unnecessary `pass`

(PIE790)